runtime/tests/interpreter/resources_test.go

```diff
@@ -2658,205 +2506,6 @@ func TestInterpretDefaultDestroyEventArgumentScoping(t *testing.T) {
-                destroy() {
-                    // Nested resource is moved here once
-                    var a <- self.a
-
-                    // Nested resource is again moved here. This one should fail.
-                    self.swap()
-
-                    destroy a
-                    destroy self.b
-                }
-            }
-
-            pub resource InnerResource {}
-
-            pub fun main() {
-                let a <- create OuterResource()
-                destroy a
-            }`,
-		)
-
-		_, err := inter.Invoke("main")
-		RequireError(t, err)
-		require.ErrorAs(t, err, &interpreter.UseBeforeInitializationError{})
-	})
-
-	t.Run("second transfer", func(t *testing.T) {
-		t.Parallel()
-
-		inter := parseCheckAndInterpret(t, `
-            pub resource OuterResource {
-                pub var a: @InnerResource
-                pub var b: @InnerResource
-
-                init() {
-                     self.a <- create InnerResource()
-                     self.b <- create InnerResource()
-                }
-
-                pub fun swap() {
-                    self.a <-> self.b
-                }
-
-                destroy() {
-                    var a <- create InnerResource()
-
-                    // Nested resource is moved here once
-                    var temp <- a <- self.a
-
-                    // Nested resource is again moved here. This one should fail.
-                    self.swap()
-
-                    destroy a
-                    destroy temp
-                    destroy self.b
-                }
-            }
-
-            pub resource InnerResource {}
-
-            pub fun main() {
-                let a <- create OuterResource()
-                destroy a
-            }`,
-		)
-
-		_, err := inter.Invoke("main")
-		RequireError(t, err)
-		require.ErrorAs(t, err, &interpreter.UseBeforeInitializationError{})
-	})
-}
-
-func TestInterpretSetMemberResourceLoss(t *testing.T) {
-
-	t.Parallel()
-
-	inter := parseCheckAndInterpret(t, `
-    access(all) resource R {
-        access(all) let id: String
-        init(id: String) {
-            self.id = id
-        }
-    }
-    access(all) fun putBack(_ rl: &ResourceLoser, _ r: @R) {
-        rl.inner <-! r
-    }
-    access(all) resource ResourceLoser {
-        pub(set) var inner: @R?;
-        init(toLose: @R) {
-            self.inner <- toLose
-        }
-        destroy() {
-            var ref = &self as &ResourceLoser;
-            // Let's move self.inner out
-            var a <- self.inner
-            // And force-write it back
-            putBack(ref, <- a!)
-        }
-    }
-    access(all) fun main(): Void {
-       var resource <- create R(id: "abc");
-       var rl <- create ResourceLoser(toLose: <- resource);
-       destroy rl
-    }`,
-	)
-
-	_, err := inter.Invoke("main")
-	RequireError(t, err)
-	require.ErrorAs(t, err, &interpreter.DestroyedResourceError{})
-}
-
-func TestInterpretValueTransferResourceLoss(t *testing.T) {
```

Comment:
Moved to runtime tests, since the custom destructors are not allowed, destruction of a resource can only be asserted from the emitted event.